parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordWriter.java

```diff
@@ -142,13 +145,23 @@ private void checkBlockSizeReached() throws IOException {
         LOG.info("mem size {} > {}: flushing {} records to disk.", memSize, nextRowGroupSize, recordCount);
         flushRowGroupToStore();
         initStore();
-        recordCountForNextMemCheck = min(max(MINIMUM_RECORD_COUNT_FOR_CHECK, recordCount / 2), MAXIMUM_RECORD_COUNT_FOR_CHECK);
+        if (estimateNextBlockSizeCheck) {
```

Comment:
Do you need this setting, or is it included to match the page size checking? I don't think we would ever want to set it to true because it is important to be under the max page size.

Reply:
Will do.

Reply:
Working on the comments @rdblue... do you think instead of a min/max record count for row group size checks, we should check every n rows?

Reply:
I've taken a closer look at this code. The only change with the `estimateNextBlockSizeCheck` is that it uses a static `min` number of records before performing the next size check if disabled.

If `estimateNextBlockSizeCheck` is enabled, then existing behavior is preserved where we try to estimate when to perform the next size check by performing the check at the halfway point of the predicted number of records (unless it's larger than the max).

So, I added the `estimateNextBlockSizeCheck` to keep it consistent with page size checking.

Reply:
Consistency here is fine. I'm not sure we will use it, but I can certainly think of valid use cases for it.